halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/plugins/AddPluginCommand.java

```diff
@@ -33,23 +35,33 @@
   private String shortDescription = "Add a plugin";
 
   @Parameter(
-      names = "--manifest-location",
-      description = "The location of the plugin's manifest file.",
-      required = true)
-  private String manifestLocation;
+      names = "--ui-resource-location",
+      description = "The location of the plugin's ui resource.",
+      required = false)
+  private String uiResourceLocation;
+
+  @Parameter(names = "--version", description = "The plugin version to use", required = false)
+  private String version;
 
   @Parameter(names = "--enabled", description = "To enable or disable the plugin.")
   private String enabled;
 
+  @Parameter(names = "--extensions", description = "A comma separated list of extensions to enable")
+  private String extensions;
+
   @Override
   protected void executeThis() {
     String currentDeployment = getCurrentDeployment();
     String name = getPluginName();
     Plugin plugin =
         new Plugin()
-            .setName(name)
+            .setId(name)
             .setEnabled(isSet(enabled) ? Boolean.parseBoolean(enabled) : false)
-            .setManifestLocation(manifestLocation);
+            .setUiResourceLocation(isSet(uiResourceLocation) ? uiResourceLocation : null)
+            .setVersion(version);
+    Arrays.stream(extensions.split(","))
+        .distinct()
```

Comment:
Do we need a `distinct` here if we're about to put the result as the key of a `Map`?

Reply:
probably not, would potentially reduce the number of objects to be GC'ed, in the case of extensions being added, but I don't see that having a big effect either way 👍